docs/guides/chip_tool_guide.md

```diff
@@ -0,0 +1,769 @@
+
+The CHIP Tool (`chip-tool`) is a tool that allows to commission a Matter device
+into the network and to communicate with it using the Zigbee Cluster Library
+(ZCL) messages.
+
+<hr>
+
+-   [Source files](#source)
+-   [Building chip-tool](#building)
+-   [Using chip-tool for Matter accessory testing](#using)
+-   [Supported commands and options](#commands)
+
+<hr>
+
+<a name="source"></a>
+
+## Source files
+
+You can find source files of the CHIP Tool in the `examples/chip-tool`
+directory.
+
+<hr>
+
+<a name="building"></a>
+
+## Building
+
+Before you can use the `chip-tool`, you must compile it from the source on Linux
+(amd64 / aarch64) or macOS.
+
+> To ensure compatibility, build the `chip-tool` and the Matter device from the
+> same revision of the connectedhomeip repository.
+
+To build and run the `chip-tool`:
+
+1. Install all necessary packages and prepare the build system. For more
+   details, see the [Building Matter](BUILDING.md) documentation:
+
+    ```
+    sudo apt-get update
+    sudo apt-get upgrade
+
+    sudo apt-get install git gcc g++ python pkg-config libssl-dev libdbus-1-dev libglib2.0-dev libavahi-client-dev ninja-build python3-venv python3-dev python3-pip unzip libgirepository1.0-dev libcairo2-dev bluez
+    ```
+
+    If the `chip-tool` is built on a Raspberry Pi, install additional packages
+    and reboot the device:
+
+    ```
+    sudo apt-get install pi-bluetooth
+    sudo reboot
+    ```
+
+2. Clone the Project CHIP repository:
+
+    ```
+    git clone https://github.com/project-chip/connectedhomeip.git
+    ```
+
+3. Enter the `connectedhomeip` directory:
+
+    ```
+    cd connectedhomeip
+    ```
+
+4. Initialize the git submodules:
+
+    ```
+    git submodule update --init
+    ```
+
+5. Run the following command:
+
+    ```
+    ./scripts/examples/gn_build_example.sh examples/chip-tool BUILD_PATH
+    ```
+
+    `BUILD_PATH` specifies where the target binaries shall to be placed.
+
+6. To run the tool execute the following command:
+
+    ```
+    ./BUILD_PATH/chip-tool
+    ```
+
+    As a result, a list of available clusters will show up. Each cluster can be
+    used as a root of the new command corresponding to ZCL data model
+    interaction, for instance:
+
+    ```
+    ./BUILD_PATH/chip-tool pairing
+    ```
+
+    will print the list of sub-operations available for `pairing` cluster.
+
+    The following is one of the possible final commands which requires
+    particular command line arguments:
+
+    ```
+    ./BUILD_PATH/chip-tool pairing ble-thread
+    ```
+
+    Execution of above line will print an error and hint the proper list of
+    parameters which must be provided to run ZCL operation.
+
+<hr>
+
+<a name="using"></a>
+
+## Using `chip-tool` for Matter accessory testing
+
+This section describes how to use `chip-tool` to test the Matter accessory.
+Below steps depend on the application clusters that you implemented on the
+device side and may be different for your accessory.
```

Comment:
Shouldn't we link to Configuring examples here?

Reply:
@greg-fer can please you point out the exact docs/link you are referring to?